react/src/components/atoms/forms/InputNumber/index.js

```diff
@@ -98,21 +98,21 @@ const NumberInput = (props) => {
               direction = 'down';
             }
             const inputEl = ref.current;
-            let newValue = inputEl.value ? Number(inputEl.value) : inputEl.value;
+            let newValue = !is.number(Number(inputEl.value)) ? 0 : Number(inputEl.value);
             if (direction === 'up' && (!hasNumberProperty(props, 'max') || newValue < props.max)) {
               // Since to Fixed returns a string, we have to cast it back to a Number
-              newValue = newValue ? Number((newValue + props.step).toFixed(countDecimals(props.step))) : props.step;
+              newValue = Number((newValue + props.step).toFixed(countDecimals(props.step)));
```

Comment:
same here since this could be null
```suggestion
              newValue = newValue ? Number((newValue + props.step).toFixed(countDecimals(props.step))) : props.step;
```